File: src/main/kotlin/santa/ping/Ping.kt
package santa.ping

import cpw.mods.fml.common.Mod
import cpw.mods.fml.common.event.FMLInitializationEvent
import cpw.mods.fml.common.event.FMLPreInitializationEvent
import net.minecraftforge.common.MinecraftForge

@Mod(name = "Ping!", modid = "ping", version = "1.0.0") class Ping {
    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        Config.initialize(event)
        if (!isSoundTypeValid(Config.soundType.toLowerCase())) {
            Config.soundType = "blop"
        }
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {
        MinecraftForge.EVENT_BUS.register(PingEventHandler())
    }

    /**
     * @param sound The sound type string provided by the configuration file.
     */
    fun isSoundTypeValid(sound: String): Boolean {
        val validTypes = arrayOf("blop", "gum", "pling", "ping", "ting", "pop")
        if (validTypes.contains(sound)) {
            return true
        } else {
            return false
        }
    }
}

File: src/main/scala/santa/ping/Ping.scala
package santa.ping

import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.{FMLInitializationEvent, FMLPreInitializationEvent}

@Mod(name = "Ping!", modid = "chatping", version = "3.0.9", acceptedMinecraftVersions = "[1.9,1.12.2]", modLanguage = "scala", clientSideOnly = true)
object Ping {
  var sound: PingSound = _
  var config: Config = _

  @Mod.EventHandler
  def preInit(event: FMLPreInitializationEvent): Unit = {
    config = new Config(event)
    sound = new PingSound(config.soundType)
  }

  @Mod.EventHandler
  def init(event: FMLInitializationEvent): Unit = {
    MinecraftForge.EVENT_BUS.register(new PingEventHandler())
  }
}
